aiida/manage/manager.py

```diff
@@ -163,6 +163,9 @@ def _load_backend(self, schema_check: bool = True, repository_check: bool = True
         # are taken into account and the `DbLogHandler` is configured.
         configure_logging(with_orm=True)
 
+        if self._backend is None:
+            raise ValueError('Could not load the backend.')
```

Comment:
Would be better to put this check directly after the logic that is supposed to set it, i.e. before the `configure_logging` call. Second, I don't think a `ValueError` is the correct exception for this. That is reserved for validation arguments of a method or function. What seems to be happening in this case, the only reason `self._backend` can be `None` is because the `backend_type` is neither `BACKEND_DJANGO` nor `BACKEND_SQLA`. So really we shouldn't have this check but add the `else` clause and raise an `AssertionError` because our internal logic is supposed to set these when loading a profile and that is what we are checking here: internal consistency of program logic.

Reply:
Good point!